Sources/BBCodeParser.php

```diff
@@ -3773,7 +3776,7 @@ protected function detectTag(string $tag_character): array
 				}
 
 				// Maybe we just want a /...
-				if ($possible['type'] == 'closed' && $next_c != ']' && substr($this->message, $this->pos + 1 + $pt_strlen, 2) != '/]' && substr($this->message, $this->pos + 1 + $pt_strlen, 3) != ' /]') {
+				if ($possible['type'] == 'closed' && $next_c != ']' && substr($this->message, (int) ($this->pos + 1 + $pt_strlen), 2) != '/]' && substr($this->message, (int) ($this->pos + 1 + $pt_strlen), 3) != ' /]') {
```

Comment:
Casting the result of the expression to int is unnecessary. The result of the arithmetic operation will already be an integer.